lib/vrl/parser/src/lex.rs

```diff
@@ -62,6 +65,7 @@ impl DiagnosticError for Error {
             Literal { .. } => 208,
             EscapeChar { .. } => 209,
             UnexpectedParseError(..) => 210,
+            UnterminatedLiteral { .. } => 211,
```

Comment:
Should we add this to the VRL error docs?

Reply:
> It actually causes the case in #6875 to no longer hang too

That's a surprise! I'll look into that.

It doesn't seem that that error number surfaces in the UI, a 202 parse error gets reported.